src/pages/studyView/charts/ChartContainer.tsx

```diff
@@ -354,6 +359,26 @@ export class ChartContainer extends React.Component<IChartContainerProps, {}> {
     get chart() {
         switch (this.chartType) {
             case ChartTypeEnum.PIE_CHART: {
+                let pieData = this.props.promise.result;
+
+                if (
```

Comment:
This condition is not valid when the initial call is already filtered, so the dataset will be smaller. But when user resets the filter, you will not get the color anymore for values that previous filtered out.